src/Components/Blazor/WebAssembly.Authentication/src/WebAssemblyAuthenticationServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,126 @@
+        /// </summary>
+        /// <typeparam name="TRemoteAuthenticationState">The state to be persisted across authentication operations.</typeparam>
+        /// <typeparam name="TProviderOptions">The configuration options of the underlying provider being used for handling the authentication operations.</typeparam>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the services to.</param>
+        /// <param name="configure">An action that will configure the <see cref="RemoteAuthenticationOptions{TProviderOptions}"/>.</param>
+        /// <returns>The <see cref="IServiceCollection"/> where the services were registered.</returns>
+        public static IServiceCollection AddSpaAuthentication<TRemoteAuthenticationState, TProviderOptions>(this IServiceCollection services, Action<RemoteAuthenticationOptions<TProviderOptions>> configure)
+            where TRemoteAuthenticationState : RemoteAuthenticationState
+            where TProviderOptions : new()
+        {
+            services.AddSpaAuthentication<RemoteAuthenticationState, TProviderOptions>();
+            if (configure != null)
+            {
+                services.Configure(configure);
+            }
+
+            return services;
+        }
+
+        /// <summary>
+        /// Adds support for authentication for SPA applications using <see cref="OidcProviderOptions"/> and the <see cref="RemoteAuthenticationState"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the services to.</param>
+        /// <param name="configure">An action that will configure the <see cref="RemoteAuthenticationOptions{TProviderOptions}"/>.</param>
+        /// <returns>The <see cref="IServiceCollection"/> where the services were registered.</returns>
+        public static IServiceCollection AddOidcAuthentication(this IServiceCollection services, Action<RemoteAuthenticationOptions<OidcProviderOptions>> configure)
+        {
+            AddSpaAuthentication<RemoteAuthenticationState, OidcProviderOptions>(services, configure);
+
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<IPostConfigureOptions<RemoteAuthenticationOptions<OidcProviderOptions>>, DefaultOidcOptionsConfiguration>());
+
+            if (configure != null)
+            {
+                services.Configure(configure);
+            }
+
+            return services;
+        }
+
+        /// <summary>
+        /// Adds support for authentication for SPA applications using <see cref="ApiAuthorizationProviderOptions"/> and the <see cref="RemoteAuthenticationState"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the services to.</param>
+        /// <returns>The <see cref="IServiceCollection"/> where the services were registered.</returns>
+        public static IServiceCollection AddApiAuthorization(this IServiceCollection services)
```

Comment:
This name also seems bad, as the term "Api" here doesn't provide any information unless you already know the history of naming within ASP.NET Core.

I know you did this for consistency with things we've had historically, but I'd still like to consider alternatives that convey more meaning. It seems unfair to expect developers to instinctively know the difference between "authorization", "SPA authorization", and "API authorization". The only name here that really conveys a clear role is "OIDC authorization".

If I'm interpreting the role of this correctly, how about `AddIdentityAuthorization`? Or is it something different?